src/BSON/Javascript.c

```diff
@@ -215,6 +216,41 @@ PHP_METHOD(Javascript, getScope)
 }
 /* }}} */
 
+/* {{{ proto array Javascript::jsonSerialize()
+*/
+PHP_METHOD(Javascript, jsonSerialize)
+{
+	php_phongo_javascript_t *intern;
+	bool has_scope;
+
+	if (zend_parse_parameters_none() == FAILURE) {
+		return;
+	}
+
+	intern = Z_JAVASCRIPT_OBJ_P(getThis());
+
+	has_scope = (intern->scope && intern->scope->len);
+	array_init_size(return_value, has_scope ? 2 : 1);
+	ADD_ASSOC_STRINGL(return_value, "$code", intern->code, intern->code_len);
```

Comment:
Works for me, I'll make a PR for [HHVM-266](https://jira.mongodb.org/browse/HHVM-266) soon.